tests/capture/test_operators.py

```diff
@@ -0,0 +1,400 @@
+    assert eqn.params == {"pauli_word": "XY", "id": None, "n_wires": 2}
+
+    assert len(eqn.invars) == 3
+    assert jaxpr.jaxpr.invars == eqn.invars
+
+    with qml.queuing.AnnotatedQueue() as q:
+        jax.core.eval_jaxpr(jaxpr.jaxpr, jaxpr.consts, 2.5, 3, 4)
+
+    assert len(q) == 1
+    assert qml.equal(q.queue[0], qml.PauliRot(2.5, "XY", (3, 4)))
+
+
+class TestTemplates:
+    def test_variable_wire_non_parametrized_template(self):
+        """Test capturing a variable wire, non-parametrized template like GroverOperator."""
+
+        jaxpr = jax.make_jaxpr(qml.GroverOperator)(wires=(0, 1, 2, 3, 4, 5))
+
+        assert len(jaxpr.eqns) == 1
+        eqn = jaxpr.eqns[0]
+
+        assert eqn.primitive == qml.GroverOperator._primitive
+        assert eqn.params == {"n_wires": 6}
+        assert eqn.invars == jaxpr.jaxpr.invars
+
+    def test_nested_template(self):
+        """Test capturing a template that contains a nested opeartion defined outside the qfunc."""
+
+        coeffs = [0.25, 0.75]
+        ops = [qml.X(0), qml.Z(0)]
+        H = qml.dot(coeffs, ops)
+
+        def qfunc(Hi):
+            qml.TrotterProduct(Hi, time=2.4, order=2)
+
+        jaxpr = jax.make_jaxpr(qfunc)(H)
+
+        assert len(jaxpr.eqns) == 6
+
+        # due to flattening and unflattening H
+        assert jaxpr.eqns[0].primitive == qml.X._primitive
+        assert jaxpr.eqns[1].primitive == qml.ops.SProd._primitive
+        assert jaxpr.eqns[2].primitive == qml.Z._primitive
+        assert jaxpr.eqns[3].primitive == qml.ops.SProd._primitive
+        assert jaxpr.eqns[4].primitive == qml.ops.Sum._primitive
+        assert not any(isinstance(eqn.outvars[0], jax.core.DropVar) for eqn in jaxpr.eqns[:5])
+
+        eqn = jaxpr.eqns[5]
+        assert eqn.primitive == qml.TrotterProduct._primitive
+        assert eqn.invars == jaxpr.eqns[4].outvars  # the sum op
+
+        assert eqn.params == {"order": 2, "time": 2.4}
+
+        with qml.queuing.AnnotatedQueue() as q:
+            jax.core.eval_jaxpr(jaxpr.jaxpr, jaxpr.consts, coeffs[0], coeffs[1])
+
+        assert len(q) == 1
+        assert q.queue[0] == qml.TrotterProduct(H, time=2.4, order=2)
+
+
+class TestOpmath:
+    """Tests for capturing operator arithmetic."""
+
+    def test_adjoint(self):
+        """Test the adjoint on an op can be captured."""
+
+        jaxpr = jax.make_jaxpr(qml.adjoint)(qml.X(0))
+
+        assert len(jaxpr.eqns) == 2
+        assert jaxpr.eqns[0].primitive == qml.X._primitive
+
+        eqn = jaxpr.eqns[1]
+        assert eqn.primitive == qml.ops.Adjoint._primitive
+        assert eqn.invars == jaxpr.eqns[0].outvars  # the pauli x op
+        assert isinstance(eqn.outvars[0].aval, AbstractOperator)
+        assert eqn.params == {}
+
+        with qml.queuing.AnnotatedQueue() as q:
+            jax.core.eval_jaxpr(jaxpr.jaxpr, jaxpr.consts)
+
+        assert len(q) == 1
+        assert qml.equal(q.queue[0], qml.adjoint(qml.X(0)))
+
+    def test_control(self):
```

Comment:
Once fixed (see other comment on `controlled.py`), we should test with `control = [3, 4]` or similar as well.

Reply:
now tested with multiple controls